selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/stack/ImportAndValidateEclipseCheProjectTest.java

```diff
@@ -0,0 +1,188 @@
+import org.eclipse.che.selenium.pageobject.Loader;
+import org.eclipse.che.selenium.pageobject.MavenPluginStatusBar;
+import org.eclipse.che.selenium.pageobject.Menu;
+import org.eclipse.che.selenium.pageobject.ProjectExplorer;
+import org.eclipse.che.selenium.pageobject.Wizard;
+import org.eclipse.che.selenium.pageobject.dashboard.CreateWorkspaceHelper;
+import org.eclipse.che.selenium.pageobject.dashboard.Dashboard;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+/** @author Aleksandr Shmaraev */
+public class ImportAndValidateEclipseCheProjectTest {
+
+  private static final String WORKSPACE_NAME = generate("EclipseCheWs", 4);
+  private static final String PROJECT_NAME = "eclipse-che";
+  private static final String ECLIPSE_CHE_PROJECT_URL = "https://github.com/eclipse/che.git";
+  private static final String PATH_TO_JAVA_FILE =
+      "selenium/che-selenium-test/src/main/java/org/eclipse/che/selenium/pageobject/CodenvyEditor.java";
+  private static final String PATH_TO_POM_FILE = "dashboard/pom.xml";
+  private static final String PATH_TO_TS_FILE = "dashboard/src/app/index.module.ts";
+
+  @Inject private Ide ide;
+  @Inject private Menu menu;
+  @Inject private ImportProjectFromLocation importProject;
+  @Inject private Loader loader;
+  @Inject private ProjectExplorer projectExplorer;
+  @Inject private Wizard projectWizard;
+  @Inject private MavenPluginStatusBar mavenPluginStatusBar;
+  @Inject private InformationDialog informationDialog;
+  @Inject private CodenvyEditor editor;
+  @Inject private Dashboard dashboard;
+  @Inject private CreateWorkspaceHelper createWorkspaceHelper;
+  @Inject private TestWorkspaceServiceClient workspaceServiceClient;
+  @Inject private DefaultTestUser defaultTestUser;
+
+  @BeforeClass
+  public void prepare() {
+    dashboard.open();
+    createWorkspaceHelper.createWorkspaceFromStackWithoutProject(ECLIPSE_CHE, WORKSPACE_NAME);
+
+    ide.switchToIdeAndWaitWorkspaceIsReadyToUse();
+    ide.waitOpenedWorkspaceIsReadyToUse();
+  }
+
+  @AfterClass
+  public void tearDown() throws Exception {
+    workspaceServiceClient.delete(WORKSPACE_NAME, defaultTestUser.getName());
+  }
+
+  @Test
+  public void checkImportAndResolveDependenciesEclipceCheProject() {
+    // import the eclipse-che project
+    projectExplorer.waitProjectExplorer();
+    menu.runCommand(WORKSPACE, IMPORT_PROJECT);
+    importProject.waitAndTypeImporterAsGitInfo(ECLIPSE_CHE_PROJECT_URL, PROJECT_NAME);
+
+    // config the project
+    projectWizard.waitCreateProjectWizardForm();
+    projectWizard.selectTypeProject(MAVEN);
+    projectWizard.clickSaveButton();
+    loader.waitOnClosed();
+
+    // TODO it is the workaround, delete it after resolve the issue
+    // TODO https://github.com/eclipse/che/issues/10515
+    // close unexpected error information dialog and click on the 'Save' button;
+    int counterErrorDialog = 0;
```

Comment:
please encapsulate this logic to the method for more readability

Reply:
Even more. Better to use the next construction:
```
webDriverWaitFactory.get().until((ExpectedCondition<Boolean>) driver -> {
      informationDialog.waitFormToOpen();
      informationDialog.clickOkBtn();
      projectWizard.clickSaveButton();
      loader.waitOnClosed();
      return informationDialog.waitFormIsOpened();
    });
```
In this case we explicitely provide time for event waiting. 
In case of using "while" the infinite loop is possible.